lonestar/libdistbench/include/DistBench/Input.h

```diff
@@ -116,6 +116,10 @@ extern cll::opt<std::string> mastersFile;
  * Graph-loading functions
  ******************************************************************************/
 
+template <typename NodeData, typename EdgeData>
+using DistGraphPtr =
```

Comment:
This using typedef is defined in CuspPartitioner.h which this file includes. The conflict of names doesn't seem to be causing issues, but it's something to keep in mind. You might be able to just reuse the one defined in CusPPartitioner.h to avoid redeclaration.

Reply:
They are just for brevity. I'll use `auto` to replace the outer layers and only keep it in where the core function resides.

Reply:
I wouldn't use auto if you can just be explicit about it; makes the code way easier to understand.

What I'm saying is that you don't need to declare DistGraphPointer again because it's already declared in CuSPPartitioner; you can just reuse it if I understand correctly.

Reply:
That's true to avoid `auto` abuse; I just wouldn't like to create dependencies on included files by the typedef, since redeclaration does not irritate the compiler.

I prefer to keep this redundancy then, which can definitely be removed later.

Reply:
Works for me. @roshandathathri any comments?

Reply:
LGTM